### PR TITLE
Asarje/resnet highres 20240624

### PR DESCRIPTION
**Pull request by mywoodstock**
### Ticket
- Link to Github Issue.

### Problem description
- Provide context for the problem.

### What's changed
- Describe the approach used to solve the problem.
- Summarize the changes made and its impact.

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: cmake/tracy.cmake [hunk 1]
```diff
@@ -28,7 +28,7 @@ ExternalProject_Add(
     INSTALL_COMMAND
         cp ${TRACY_HOME}/csvexport/build/unix/csvexport-release .
     BUILD_COMMAND
-        cd ${TRACY_HOME}/csvexport/build/unix && TRACY_NO_LTO=1 make -f ${TRACY_HOME}/csvexport/build/unix/Makefile
+        cd ${TRACY_HOME}/csvexport/build/unix && CXX=g++ TRACY_NO_LTO=1 make -f ${TRACY_HOME}/csvexport/build/unix/Makefile
```
Comment (tt-rkim) [inline cmake/tracy.cmake:31]: @mo-tenstorrent @vtangTT 
Are we aware of this? This seems like a notable change

**Review by boris-drazic — APPROVED** [review]

File: cmake/tracy.cmake (hunk 1 above)
Comment (mywoodstock) [reply cmake/tracy.cmake:31]: this will not go in from my side :) i will remove it from this PR.

Comment (vtangTT) [reply cmake/tracy.cmake:31]: I have an old PR [here](https://github.com/tenstorrent/tt-metal/pull/9425), but @TT-billteng and I didn't want to force tracy to use g++ while we are using Clang for metal. 

The current plan is the bump Tracy tools to use CMakeLists ([issue](https://github.com/tenstorrent/tt-metal/issues/9691))so we have better control over the build env. Encountering a number of build problems but I will be actively working on this these next few days and hope to get it in by end of week. 

I don't want to block your [PR](https://github.com/tenstorrent/tt-metal/pull/9660), so we could submit the workaround for now? @TT-billteng

Comment (mo-tenstorrent) [reply cmake/tracy.cmake:31]: @mywoodstock without this your builds break on ird right?

Comment (mywoodstock) [reply cmake/tracy.cmake:31]: Yes, unable to build on IRD without this hack.
@vtangTT @TT-billteng if this workaround is ok, i can keep it with this PR. LMK.

Comment (TT-billteng) [reply cmake/tracy.cmake:31]: let's submit this for now and try to fix it with the Tracy submodule bump

File: tt_eager/tt_dnn/op_library/conv/optimized_conv_op.cpp [hunk 2]
```diff
@@ -354,6 +354,17 @@ std::vector<Shape> OptimizedConvNew::compute_output_shapes(const std::vector<Ten
...
+    uint32_t num_cores_x = grid_size.x;
+    uint32_t num_cores_y = grid_size.y;
+    uint32_t total_available_cores = num_cores_x * num_cores_y;
+    num_cores_along_width = num_cores_along_width > 0 ? num_cores_along_width : num_cores_x;
+    TT_FATAL(target_num_cores % num_cores_along_width == 0, "The grid is not a rectangle: target_num_cores = {}, num_cores_along_width = {}", target_num_cores, num_cores_along_width);
+    TT_FATAL(target_num_cores <= total_available_cores, "Target number of cores {} is greater than total number of available cores {}", target_num_cores, total_available_cores);
+    CoreRange core_range = {{0, 0}, {target_num_cores / num_cores_along_width - 1, num_cores_along_width - 1}};
+    return CoreRangeSet({core_range});
+}
+
```
Comment (TT-BrianLiu) [inline tt_eager/tt_dnn/op_library/conv/optimized_conv_op.cpp:367]: What is the point of this function?

File: tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet50_large_new.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        super().__init__()
+        torch.manual_seed(0)
+        self.pcc_passed = False
+        self.pcc_message = "Did you forget to call validate()?"
+        self.device = device
+        self.batch_size = batch_size
+        self.act_dtype = act_dtype
+        self.weight_dtype = weight_dtype
+        self.math_fidelity = math_fidelity
+
```
Comment (TT-BrianLiu) [inline tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet50_large_new.py:188]: Can this be reused for all the tests?

**Review by TT-BrianLiu — COMMENTED** [review]
Some questions:
- There seems to be a lot of code for models + testing. Is there anyway we can reuse more code or is it all completely different?
- What is `OptimizedConvNew` and how is it different from `OptimizedConv`?

**Review by tt-rkim — APPROVED** [review]

File: tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet50_large_new.py (hunk 3 above)
Comment (mywoodstock) [reply tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet50_large_new.py:188]: yes can be used for all resnet tests -> i plan to consolidate this for all high-res variants .. but later PR

File: tt_eager/tt_dnn/op_library/conv/optimized_conv_op.cpp (hunk 2 above)
Comment (mywoodstock) [reply tt_eager/tt_dnn/op_library/conv/optimized_conv_op.cpp:367]: The generic version does not calculate the grid for block sharding correctly (with x != grid width)

**Review by tt-nshanker — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]

**Pull request merged**